### DISCUSSION TITLE
wingetcreate update - handle is invalid

### DISCUSSION THREAD

**Discussion by cdburke** [microsoft/winget-create#369] [Q&A]
I have a package listed by winget search J-language as:

  J-language Jsoftware.J.9.5 9.5.0-beta1 winget

Trying to update it gives an error message:

PS C:\Users\elmo> wingetcreate.exe update Jsoftware.J.9.5 --version 9.5.0-beta2 --interactive
Retrieving latest manifest for Jsoftware.J.9.5
System.IO.IOException: The handle is invalid.
   at System.ConsolePal.GetBufferInfo(Boolean throwOnNoConsole, Boolean& succeeded)
   at System.ConsolePal.Clear()
   at Microsoft.WingetCreateCLI.Commands.UpdateCommand.UpdateManifestsInteractively(Manifests manifests)
   at Microsoft.WingetCreateCLI.Commands.UpdateCommand.ExecuteManifestUpdate(List`1 latestManifestContent, CommandExecutedEvent commandEvent)
   at Microsoft.WingetCreateCLI.Commands.UpdateCommand.Execute()
   at Microsoft.WingetCreateCLI.Program.Main(String[] args)

Is there a problem updating the version from 9.5.0-beta1 to 9.5.0-beta2?

**Comment by mdanish-kh** [comment]
Can you share the InstallerUrl you used? (You might want to convert this discussion into an issue)

**Comment by cdburke** [comment]
I was doing this interactively and had not yet got a prompt for the
InstallerlUrl.

However, thinking about it I realize a possible problem. For the
InstallerUrl I used

   https://www.jsoftware.com/download/j9.5/install/j9.5_win64.exe

which is a symlink to the latest version, i.e. previously:

   https://www.jsoftware.com/download/j9.5/install/j9.5.0-beta1_win64.exe

and now updated to:

   https://www.jsoftware.com/download/j9.5/install/j9.5.0-beta2_win64.exe

Perhaps wingetcreate saw that the previous InstallerUrl had changed? So I
should not use a symlink that is updated for it?

**Reply by mdanish-kh** [reply]
I could not reproduce the issue on my end; I used the same command and was prompted for the interactive update 
![image](https://user-images.githubusercontent.com/88161975/235989714-414d437b-17ed-41ba-b0ec-b178c77d888b.png)
Can you share what version of winget-create you are running by just typing in `wingetcreate` in the terminal?

**Comment by cdburke** [comment]
c:\users\elmo>wingetcreate
Windows Package Manager Manifest Creator v1.2.5.0

**Comment by cdburke** [comment]
Very strange. I tried this on another machine, and got exactly the same
problem!
